cleanrl/ddpg_continuous_action.py

```diff
@@ -167,7 +176,11 @@ def forward(self, x):
                 [
                     (
                         actions.tolist()[0]
-                        + np.random.normal(0, max_action * args.exploration_noise, size=envs.single_action_space.shape[0])
+                        + np.random.normal(
+                            actor.action_bias[0].cpu().numpy(),
```

Comment:
Could you change this part? I don’t think we should transfer the data back to CPU every time there is an update.

Reply:
How is this instead ? The noise sampling is done at GPU level where the actions already are, then the action is moved to CPU where it is clipped before being paseed to the env. Looks even cleaner.

```python
            actions = actor(torch.Tensor(obs).to(device))
            actions += torch.normal(actor.action_bias, actor.action_scale * args.exploration_noise)
            actions = actions.detach().cpu().numpy().clip(envs.single_action_space.low, envs.single_action_space.high)
```

Reply:
I think that sounds good. Thank you @dosssman